### PR TITLE
feat: Experimental warning dialog

### LINKED ISSUE #2793: Add a dialog notifying that the app is prototype
Add popup screen: "This mobile app is still in its prototype stage with many features being experimental. We currently recommend using this app only for testing."

The popup should only display once

### PR DESCRIPTION
**Pull request by nileshteji**
Fixes #2793 

Changes: Preference Added to detect when should be the dialog showed with a suitable message.

Screenshots for the change:

[Google Drive](https://drive.google.com/file/d/1msRv4lz7-Q0Cx7r5hht3bx3b03ZyukoM/view?usp=sharing)

### CODE REVIEW

**Comment by nileshteji** [comment]
@iamareebjamal Can you review the PR Sir ?

File: app/src/main/java/org/fossasia/openevent/general/welcome/WelcomeFragment.kt [hunk 1]
```diff
@@ -31,17 +31,37 @@ class WelcomeFragment : Fragment() {
...
+                }.show()
+            else -> TODO()
+
+        }
+
+
+
+
+
+
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/general/welcome/WelcomeFragment.kt]: Why this many blank lines?

File: app/src/main/java/org/fossasia/openevent/general/welcome/WelcomeFragment.kt [hunk 2]
```diff
@@ -31,17 +31,37 @@ class WelcomeFragment : Fragment() {
...
     val preference = Preference()
+    private lateinit var editor: SharedPreferences.Editor
+
 
     override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
         rootView = inflater.inflate(R.layout.fragment_welcome, container, false)
         val thisActivity = activity
         if (thisActivity is AppCompatActivity)
             thisActivity.supportActionBar?.hide()
+
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/general/welcome/WelcomeFragment.kt]: Why this change?

File: app/src/main/java/org/fossasia/openevent/general/welcome/WelcomeFragment.kt [hunk 3]
```diff
@@ -31,17 +31,37 @@ class WelcomeFragment : Fragment() {
...
         rootView.pickCityButton.setOnClickListener {
             Navigation.findNavController(rootView)
                 .navigate(WelcomeFragmentDirections.actionWelcomeToSearch(WELCOME_FRAGMENT))
         }
 
+
+        when (preference.getBoolean("dialog", false)) {
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/general/welcome/WelcomeFragment.kt]: Pick better preference name. `dialog` is too generic

File: app/src/main/java/org/fossasia/openevent/general/welcome/WelcomeFragment.kt [hunk 4]
```diff
@@ -31,17 +31,37 @@ class WelcomeFragment : Fragment() {
...
+            false -> AlertDialog.Builder(thisActivity!!)
+                .setTitle("Message")
+                .setMessage(R.string.dialog_message).setPositiveButton("Okay") { dialog, it ->
+                    preference.putBoolean("dialog",false)
+                    dialog.dismiss()
+                }.show()
+            else -> TODO()
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/general/welcome/WelcomeFragment.kt]: So, you want to crash the app if preference is true?

File: app/src/main/java/org/fossasia/openevent/general/welcome/WelcomeFragment.kt [hunk 5]
```diff
@@ -31,17 +31,37 @@ class WelcomeFragment : Fragment() {
...
             Navigation.findNavController(rootView)
                 .navigate(WelcomeFragmentDirections.actionWelcomeToSearch(WELCOME_FRAGMENT))
         }
 
+
+        when (preference.getBoolean("dialog", false)) {
+            false -> AlertDialog.Builder(thisActivity!!)
+                .setTitle("Message")
+                .setMessage(R.string.dialog_message).setPositiveButton("Okay") { dialog, it ->
+                    preference.putBoolean("dialog",false)
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/general/welcome/WelcomeFragment.kt]: I don't get it. You show the dialog if the preference is false, and then set the preference to false when the user clicks OK?

File: app/src/main/java/org/fossasia/openevent/general/welcome/WelcomeFragment.kt [hunk 6]
```diff
@@ -92,7 +112,7 @@ class WelcomeFragment : Fragment() {
 
     private fun redirectToAuth() {
         Navigation.findNavController(rootView).navigate(WelcomeFragmentDirections.actionWelcomeToAuth(
-                redirectedFrom = WELCOME_FRAGMENT, showSkipButton = true)
+            redirectedFrom = WELCOME_FRAGMENT, showSkipButton = true)
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/general/welcome/WelcomeFragment.kt]: Why this change?

File: app/src/main/java/org/fossasia/openevent/general/welcome/WelcomeFragment.kt [hunk 7]
```diff
@@ -31,17 +31,37 @@ class WelcomeFragment : Fragment() {
...
         rootView.pickCityButton.setOnClickListener {
             Navigation.findNavController(rootView)
                 .navigate(WelcomeFragmentDirections.actionWelcomeToSearch(WELCOME_FRAGMENT))
         }
 
+
+        when (preference.getBoolean("dialog", false)) {
+            false -> AlertDialog.Builder(thisActivity!!)
+                .setTitle("Message")
+                .setMessage(R.string.dialog_message).setPositiveButton("Okay") { dialog, it ->
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/general/welcome/WelcomeFragment.kt]: No need of `Okay`. OK is the standard

File: app/src/main/java/org/fossasia/openevent/general/welcome/WelcomeFragment.kt (hunk 1 above)
Comment (nileshteji) [reply app/src/main/java/org/fossasia/openevent/general/welcome/WelcomeFragment.kt]: I have Reformated the code @iamareebjamal .

File: app/src/main/java/org/fossasia/openevent/general/welcome/WelcomeFragment.kt (hunk 2 above)
Comment (nileshteji) [reply app/src/main/java/org/fossasia/openevent/general/welcome/WelcomeFragment.kt]: Reformated the code

File: app/src/main/java/org/fossasia/openevent/general/welcome/WelcomeFragment.kt (hunk 3 above)
Comment (nileshteji) [reply app/src/main/java/org/fossasia/openevent/general/welcome/WelcomeFragment.kt]: The preference name changed to dialog_prompt

File: app/src/main/java/org/fossasia/openevent/general/welcome/WelcomeFragment.kt (hunk 6 above)
Comment (nileshteji) [reply app/src/main/java/org/fossasia/openevent/general/welcome/WelcomeFragment.kt]: Refractored

File: app/src/main/java/org/fossasia/openevent/general/welcome/WelcomeFragment.kt (hunk 7 above)
Comment (nileshteji) [reply app/src/main/java/org/fossasia/openevent/general/welcome/WelcomeFragment.kt]: Done

File: app/src/main/java/org/fossasia/openevent/general/welcome/WelcomeFragment.kt [hunk 8]
```diff
@@ -28,9 +27,12 @@ const val LOCATION_PERMISSION_REQUEST = 1000
 const val WELCOME_FRAGMENT = "welcomeFragment"
 
 class WelcomeFragment : Fragment() {
+    
     private lateinit var rootView: View
     private val geoLocationViewModel by viewModel<GeoLocationViewModel>()
     val preference = Preference()
+    private lateinit var editor: SharedPreferences.Editor
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/general/welcome/WelcomeFragment.kt]: Where is this used?

File: app/src/main/java/org/fossasia/openevent/general/welcome/WelcomeFragment.kt [hunk 9]
```diff
@@ -41,7 +43,14 @@ class WelcomeFragment : Fragment() {
             Navigation.findNavController(rootView)
                 .navigate(WelcomeFragmentDirections.actionWelcomeToSearch(WELCOME_FRAGMENT))
         }
-
+        if(!preference.getBoolean("dialog_prompt", false)) {
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/general/welcome/WelcomeFragment.kt]: dialog_prompt -> experimental_warning_shown

File: app/src/main/java/org/fossasia/openevent/general/welcome/WelcomeFragment.kt [hunk 10]
```diff
@@ -41,7 +43,14 @@ class WelcomeFragment : Fragment() {
...
+            AlertDialog.Builder(thisActivity!!)
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/general/welcome/WelcomeFragment.kt]: requireActivity

**Comment by nileshteji** [comment]
@iamareebjamal can you check the Approvements. Thank You

File: app/src/main/java/org/fossasia/openevent/general/welcome/WelcomeFragment.kt [hunk 11]
```diff
@@ -41,7 +43,14 @@ class WelcomeFragment : Fragment() {
...
+                .setTitle("Message")
+                .setMessage(R.string.dialog_message).setPositiveButton("OK") { dialog, it ->
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/general/welcome/WelcomeFragment.kt]: Move setPositiveButton to next line and use R.string

File: app/src/main/java/org/fossasia/openevent/general/welcome/WelcomeFragment.kt [hunk 12]
```diff
@@ -41,7 +43,14 @@ class WelcomeFragment : Fragment() {
...
+                    preference.putBoolean("dialog_prompt", true)
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/general/welcome/WelcomeFragment.kt]: Move the preference key string in a variable

File: app/src/main/java/org/fossasia/openevent/general/welcome/WelcomeFragment.kt (hunk 8 above)
Comment (iamareebjamal) [reply app/src/main/java/org/fossasia/openevent/general/welcome/WelcomeFragment.kt]: Do not resolve conversations manually. It is automatically resolved once requested changes are made

**Comment by iamareebjamal** [comment]
@nileshteji Do not resolve conversations manually. It is automatically resolved once requested changes are made

**Comment by nileshteji** [comment]
@iamareebjamal Okay Sir.

**Comment by nileshteji** [comment]
@iamanbansal Sir can you review the changes

File: app/src/main/java/org/fossasia/openevent/general/welcome/WelcomeFragment.kt [hunk 13]
```diff
@@ -26,12 +24,15 @@ import org.koin.androidx.viewmodel.ext.android.viewModel
 
 const val LOCATION_PERMISSION_REQUEST = 1000
 const val WELCOME_FRAGMENT = "welcomeFragment"
+const val PREFERENCE_KEY = "experimental_warning_shown"
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/general/welcome/WelcomeFragment.kt]:
```suggestion
const val EXPERIMENTAL_WARNING_PREFERENCE_KEY = "experimental_warning_shown"
```

Comment (iamareebjamal) [reply app/src/main/java/org/fossasia/openevent/general/welcome/WelcomeFragment.kt]: Please always use explicit variable names, not generic names like `PREFERENCE_KEY`

File: app/src/main/res/values/strings.xml [hunk 14]
```diff
@@ -537,5 +537,7 @@
     <string name="social_link">Social Link</string>
     <string name="tick">tick</string>
     <string name="general_settings_and_support">General Settings and Support</string>
+    <string name="dialog_message">This mobile app is still in its prototype stage with many features being experimental. We currently recommend using this app only for testing.</string>
+    <string name="positive_button_dialog">OK</string>
```
Comment (iamareebjamal) [inline app/src/main/res/values/strings.xml]: This must already be in the strings file. Please check

**Comment by iamareebjamal** [comment]
Also, the build is failing. Please check the reason

**Comment by nileshteji** [comment]
@iamareebjamal  The build is failing due to Task :app:spotlessKotlin FAILED which is an indentation error or due to the whitespaces. It is not affecting the functionality of the Application. Thank You [Verbose info for the Error](https://github.com/diffplug/spotless/issues/276)

**Comment by iamareebjamal** [comment]
We put that check deliberately in the CI. It is not an accident. Please fix it

File: app/src/main/java/org/fossasia/openevent/general/welcome/WelcomeFragment.kt [hunk 15]
```diff
@@ -91,8 +99,8 @@ class WelcomeFragment : Fragment() {
     }
 
     private fun redirectToAuth() {
-        Navigation.findNavController(rootView).navigate(WelcomeFragmentDirections.actionWelcomeToAuth(
-                redirectedFrom = WELCOME_FRAGMENT, showSkipButton = true)
-        )
+        Navigation.findNavController(rootView)
+            .navigate(WelcomeFragmentDirections
+                .actionWelcomeToAuth(redirectedFrom = WELCOME_FRAGMENT, showSkipButton = true))
```
Comment (nileshteji) [inline app/src/main/java/org/fossasia/openevent/general/welcome/WelcomeFragment.kt:104]: This has to change because of the build failure the check was not able to pass because the line length was greater than 120 which is one of the checks in ktlint

**Comment by nileshteji** [comment]
@iamareebjamal BUILD is Fixed. Can you check the PR?

**Review by iamareebjamal — APPROVED** [review]
